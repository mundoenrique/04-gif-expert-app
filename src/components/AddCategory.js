import { useState } from "react"

export default function AddCategory() {
	const [inputValue, setInputValue] = useState('Hola doctor');

	const handleInputChange = (e) => {
		setInputValue(e.target.value);
	}

	return (
		<>
			<input
				type="text"
				value={inputValue}
				onChange={handleInputChange}
			/>
		</>
	)
}

