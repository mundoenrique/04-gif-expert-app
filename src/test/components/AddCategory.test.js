import { shallow } from "enzyme";
import AddCategory from "../../components/AddCategory";

describe('Pruebas para el componente <AddCategory />', () => {
	const setCategories = () => { }
	const wrapper = shallow(<AddCategory setCategories={setCategories} />);

	test('Debe mostrarse correctamente', () => {

		expect(wrapper).toMatchSnapshot();
	});

	test('Debe cambiar el input', () => {
		const input = wrapper.find('input');
		const value = 'Goku';

		input.simulate('change', { target: { value } });

		const inputValue = wrapper.find('input');

		expect(inputValue.prop('value')).toBe(value)
	});
});
