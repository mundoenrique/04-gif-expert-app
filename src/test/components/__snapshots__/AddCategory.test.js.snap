// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`Pruebas para el componente <AddCategory /> Debe mostrarse correctamente 1`] = `
<Fragment>
  <form
    onSubmit={[Function]}
  >
    <input
      onChange={[Function]}
      type="text"
      value=""
    />
  </form>
</Fragment>
`;
